From 6a85df0efe0bd49ba366dd6db3639df395a1974a Mon Sep 17 00:00:00 2001
From: Robert Nelson <robertcnelson@gmail.com>
Date: Wed, 25 May 2016 13:57:50 -0500
Subject: [PATCH] sama5d4ek: uEnv.txt, bootz, n fixes

Signed-off-by: Robert Nelson <robertcnelson@gmail.com>
---
 include/configs/at91-sama5_common.h    | 9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)

diff --git a/include/configs/at91-sama5_common.h b/include/configs/at91-sama5_common.h
index dc955b2..f9b5786 100644
--- a/include/configs/at91-sama5_common.h
+++ b/include/configs/at91-sama5_common.h
@@ -49,6 +49,8 @@
  */
 
 #ifdef CONFIG_SYS_USE_MMC
+#define CONFIG_PARTITION_UUIDS
+#define CONFIG_CMD_PART
 
 #ifdef CONFIG_ENV_IS_IN_MMC
 /* Use raw reserved sectors to save environment */
@@ -68,12 +70,11 @@
 #define CONFIG_BOOTCOMMAND	"if test ! -n ${dtb_name}; then "	\
 				    "setenv dtb_name at91-${board_name}.dtb; " \
 				"fi; "					\
-				"fatload mmc 0:1 0x21000000 ${dtb_name}; " \
+				"part uuid mmc 0:2 partuuid; " \
+				"setenv bootargs console=ttyS0,115200 root=PARTUUID=${partuuid} ro rootwait; " \
+				"fatload mmc 0:1 0x21000000 /dtbs/${dtb_name}; " \
 				"fatload mmc 0:1 0x22000000 zImage; "	\
 				"bootz 0x22000000 - 0x21000000"
-#define CONFIG_BOOTARGS							\
-	"console=ttyS0,115200 earlyprintk "				\
-	"root=/dev/mmcblk0p2 rw rootwait"
 #else
 #define CONFIG_BOOTARGS							\
 	"console=ttyS0,115200 earlyprintk "				\
-- 
2.8.1

