From f150af78e6935709328fb668857d9dc5acab7e70 Mon Sep 17 00:00:00 2001
From: Robert Nelson <robertcnelson@gmail.com>
Date: Fri, 15 Jan 2016 14:34:03 -0600
Subject: [PATCH] ls1021atwr fixes

Signed-off-by: Robert Nelson <robertcnelson@gmail.com>
---
 include/configs/ls1021atwr.h | 44 +++++++++++++++++++++++++++++++++-----------
 1 file changed, 33 insertions(+), 11 deletions(-)

diff --git a/include/configs/ls1021atwr.h b/include/configs/ls1021atwr.h
index c12ba3a..47ecf26 100644
--- a/include/configs/ls1021atwr.h
+++ b/include/configs/ls1021atwr.h
@@ -430,20 +430,41 @@
 
 #define CONFIG_FSL_DEVICE_DISABLE
 
-#define CONFIG_BOOTDELAY		3
-
 #ifdef CONFIG_LPUART
-#define CONFIG_EXTRA_ENV_SETTINGS       \
-	"bootargs=root=/dev/ram0 rw console=ttyLP0,115200\0" \
-	"initrd_high=0xffffffff\0"      \
-	"fdt_high=0xffffffff\0"
+#define CONFIG_CONSOLE_DEV	"ttyLP0"
 #else
-#define CONFIG_EXTRA_ENV_SETTINGS	\
-	"bootargs=root=/dev/ram0 rw console=ttyS0,115200\0" \
-	"initrd_high=0xffffffff\0"      \
-	"fdt_high=0xffffffff\0"
+#define CONFIG_CONSOLE_DEV	"ttyS0"
 #endif
 
+#ifndef CONFIG_SPL_BUILD
+#define CONFIG_EXTRA_ENV_SETTINGS \
+	"console=" CONFIG_CONSOLE_DEV ",115200\0" \
+	"fdtfile=ls1021a-twr.dtb\0" \
+	"fdt_high=0xffffffff\0" \
+	"initrd_high=0xffffffff\0" \
+	"fdt_addr_r=0x18000000\0" \
+	"fdt_addr=0x18000000\0" \
+	"mmcdev=" __stringify(CONFIG_SYS_MMC_ENV_DEV) "\0" \
+	"kernel_addr_r=" __stringify(CONFIG_LOADADDR) "\0"  \
+	"pxe_addr_r=" __stringify(CONFIG_LOADADDR) "\0" \
+	"ramdisk_addr_r=0x13000000\0" \
+	"ramdiskaddr=0x13000000\0" \
+	"scriptaddr=" __stringify(CONFIG_LOADADDR) "\0" \
+	BOOTENV
+
+#define CONFIG_BOOTCOMMAND \
+	"run distro_bootcmd"
+
+#define BOOT_TARGET_DEVICES(func) \
+	func(MMC, mmc, 0)
+
+#include <config_distro_defaults.h>
+#include <config_distro_bootcmd.h>
+
+#else
+#define CONFIG_EXTRA_ENV_SETTINGS
+#endif /* CONFIG_SPL_BUILD */
+
 /*
  * Miscellaneous configurable options
  */
@@ -463,7 +484,8 @@
 #define CONFIG_SYS_MEMTEST_START	0x80000000
 #define CONFIG_SYS_MEMTEST_END		0x9fffffff
 
-#define CONFIG_SYS_LOAD_ADDR		0x82000000
+#define CONFIG_LOADADDR			0x82000000
+#define CONFIG_SYS_LOAD_ADDR		CONFIG_LOADADDR
 
 #define CONFIG_LS102XA_STREAM_ID
 
-- 
2.7.0.rc3

